sdk/serialization/azure-json/src/main/java/com/azure/json/JsonWriter.java

```diff
@@ -546,14 +715,45 @@ public final <T> JsonWriter writeArrayField(String fieldName, T[] array,
      */
     public final <T> JsonWriter writeArrayField(String fieldName, Iterable<T> array,
         WriteValueCallback<JsonWriter, T> elementWriterFunc) throws IOException {
+        return writeArrayField(fieldName, array, elementWriterFunc, false);
+    }
+
+    /**
+     * Writes a JSON array field.
+     * <p>
+     * This API will begin by writing the field name and start array ({@code [}) followed by all elements in the array
+     * using the {@code elementWriterFunc} and finishing by writing the end array ({@code ]}).
+     * <p>
+     * If an element in the array is null, whether writing {@link JsonToken#NULL} for the element will be determined by
+     * the {@code elementWriterFunc}'s handling of null values. For example, {@link #writeJson(JsonSerializable)} will
+     * skip writing null values, so if that is used nothing will be written for a null element.
+     * <p>
+     * The field is only written when {@code value} isn't null, if a null field needs to be written use
+     * {@link #writeNullableField(String, Object, WriteValueCallback)}.
+     * <p>
+     * Combines {@link #writeFieldName(String)} and {@link #writeArray(Object[], WriteValueCallback, boolean)} to
+     * simplify adding a key-value to a JSON object.
+     *
+     * @param fieldName The field name.
+     * @param array The array being written.
+     * @param elementWriterFunc The function that writes each element of the array.
+     * @param explicitlyWriteNull Whether null values will explicitly be written.
+     * @param <T> The array element type.
+     * @return The updated JsonWriter object.
+     * @throws NullPointerException If {@code fieldName} or {@code elementWriterFunc} is null.
+     * @throws IOException If the {@code fieldName} or the JSON array fails to be written, either JSON start or end
+     * array or the element write.
+     */
+    public <T> JsonWriter writeArrayField(String fieldName, Iterable<T> array,
```

Comment:
In an array, should the default behavior be to always write null values and only skip when explicitly requested to skip?

Reply:
Will change the behavior to always write a JSON null in serialization unless explicitly directed to skip null values.